File: Other/2025/1979. Find Greatest Common Divisor of Array.py
from typing import List
import math

class Solution:
    def findGCD(self, nums: List[int]) -> int:
        return math.gcd(min(nums), max(nums))

# Test cases
print(Solution().findGCD([2,5,6,9,10]))    
print(Solution().findGCD([7,5,6,8,3]))    
print(Solution().findGCD([3,3]))    

File: Other/2025/1979. Find Greatest Common Divisor of Array.ts
function findGCD(nums: number[]): number {
  let a: number = Math.min(...nums);
  let b: number = Math.max(...nums);

  const euclidean = (a: number, b: number) => {
    if (b === 0) return a;
    return euclidean(b, a % b);
  };

  return euclidean(b, a);
}

// Test cases
console.log(findGCD([7, 5, 6, 8, 3]));
console.log(findGCD([3, 3]));
